invalid memory address or nil pointer dereference in message.go:30

Issue during connection with redis directly.
```
2023/05/12 22:52:53 http: panic serving [::1]:59594: runtime error: invalid memory address or nil pointer dereference
goroutine 949 [running]:
net/http.(*conn).serve.func1()
        /usr/local/go/src/net/http/server.go:1854 +0xbf
panic({0x3515200, 0x705fbd0})
        /usr/local/go/src/runtime/panic.go:890 +0x263
...
panic({0x3515200, 0x705fbd0})
        /usr/local/go/src/runtime/panic.go:884 +0x213
github.com/redis/rueidis.(*RedisError).Error(0x7077840?)
        /home/fank/go/pkg/mod/github.com/redis/rueidis@v1.0.2/message.go:30
gitlab.com/mcldev/negsoft/api/src/middleware.AuthorizationHandler.func1.1(0xc00234a900, {0x45afa90, 0xc003c161c0}, 0xc001643780)
...
```

https://github.com/redis/rueidis/blob/v1.0.2/message.go#L30 affected line.

Here a cut from how the calls to redis looks like:
```go
	permissions := []string{"foo", "bar"}
	fieldValue := redisClient.B().Hmset().Key(key).FieldValue()
	for _, v := range permissions {
		fieldValue = fieldValue.FieldValue(v, "1")
	}

	response = redisClient.DoMulti(ctx, fieldValue.Build(), redisClient.B().Expire().Key(key).Seconds(12*60*60).Build())
	for i, v := range response {
		err = v.RedisError()
		fmt.Println(i, err)
		fmt.Printf("%#v", err)
		if err == nil {
			fmt.Println("11111")
		}
		if err != nil {
			return false, err
		}

		fmt.Println("asd")
	}
```

the part of how i test it.
```go
client.EXPECT().DoMulti(ctx, mock.Match("HMSET", "core:user:permission:123", "view_posts", "1"), mock.Match("EXPIRE", "core:user:permission:123", "43200")).Return([]rueidis.RedisResult{
    mock.Result(mock.RedisString("OK")),
    mock.Result(mock.RedisInt64(1)),
})
```

the debug output in the tests:
```
=== RUN   TestGuard_CanAccessAny
--- FAIL: TestGuard_CanAccessAny (0.00s)
=== RUN   TestGuard_CanAccessAny/key_miss
    --- FAIL: TestGuard_CanAccessAny/key_miss (0.00s)
=== RUN   TestGuard_CanAccessAny/key_miss/permission_miss
        --- FAIL: TestGuard_CanAccessAny/key_miss/permission_miss (0.00s)
=== RUN   TestGuard_CanAccessAny/key_miss/permission_miss/loader_single
0 <nil>
(*rueidis.RedisError)(nil)    guard_test.go:165: 
        	Error Trace:	C:/Users/info/repo/api/src/negsoft/user_permission/guard_test.go:165
        	Error:      	Received unexpected error:
        	            	<nil>
        	Test:       	TestGuard_CanAccessAny/key_miss/permission_miss/loader_single
            --- FAIL: TestGuard_CanAccessAny/key_miss/permission_miss/loader_single (0.00s)
```

Looks like it doesn't like the combination of HMSET and EXPIRE in a DoMulti, but its just a guess still investigating.

--- a/lua.go
+++ b/lua.go
@@ -38,7 +38,7 @@ func (s *Lua) Exec(ctx context.Context, c Client, keys, args []string) (resp Red
 	} else {
 		resp = c.Do(ctx, c.B().Evalsha().Sha1(s.sha1).Numkeys(int64(len(keys))).Key(keys...).Arg(args...).Build())
 	}
-	if err := resp.RedisError(); err != nil && err.IsNoScript() {
+	if err, ok := IsRedisErr(resp.Error()); ok && err.IsNoScript() {
 		if s.readonly {
 			resp = c.Do(ctx, c.B().EvalRo().Script(s.script).Numkeys(int64(len(keys))).Key(keys...).Arg(args...).Build())
 		} else {

--- a/lua_test.go
+++ b/lua_test.go
@@ -68,7 +68,7 @@ func TestNewLuaScript(t *testing.T) {
 
 	script := NewLuaScript(body)
 
-	if !script.Exec(context.Background(), c, k, a).RedisError().IsNil() {
+	if err, ok := IsRedisErr(script.Exec(context.Background(), c, k, a).Error()); ok && !err.IsNil() {
 		t.Fatalf("ret mistmatch")
 	}
 }
@@ -101,7 +101,7 @@ func TestNewLuaScriptReadOnly(t *testing.T) {
 
 	script := NewLuaScriptReadOnly(body)
 
-	if !script.Exec(context.Background(), c, k, a).RedisError().IsNil() {
+	if err, ok := IsRedisErr(script.Exec(context.Background(), c, k, a).Error()); ok && !err.IsNil() {
 		t.Fatalf("ret mistmatch")
 	}
 }

--- a/message.go
+++ b/message.go
@@ -17,10 +17,16 @@ const messageStructSize = int(unsafe.Sizeof(RedisMessage{}))
 // Nil represents a Redis Nil message
 var Nil = &RedisError{typ: '_'}
 
-// IsRedisNil is a handy method to check if error is redis nil response.
+// IsRedisNil is a handy method to check if error is a redis nil response.
 // All redis nil response returns as an error.
 func IsRedisNil(err error) bool {
 	return err == Nil
+}
+
+// IsRedisErr is a handy method to check if error is a redis ERR response.
+func IsRedisErr(err error) (ret *RedisError, ok bool) {
+	ret, ok = err.(*RedisError)
+	return ret, ok && ret != Nil
 }
 
 // RedisError is an error response or a nil message from redis instance
@@ -82,14 +88,6 @@ func newErrResult(err error) RedisResult {
 type RedisResult struct {
 	err error
 	val RedisMessage
-}
-
-// RedisError can be used to check if the redis response is an error message.
-func (r RedisResult) RedisError() *RedisError {
-	if err := r.val.Error(); err != nil {
-		return err.(*RedisError)
-	}
-	return nil
 }
 
 // NonRedisError can be used to check if there is an underlying error (ex. network timeout).

--- a/message_test.go
+++ b/message_test.go
@@ -23,6 +23,22 @@ func TestIsRedisNil(t *testing.T) {
 	}
 }
 
+func TestIsRedisErr(t *testing.T) {
+	err := Nil
+	if ret, ok := IsRedisErr(err); ok || ret != Nil {
+		t.Fatal("TestIsRedisErr fail")
+	}
+	if ret, ok := IsRedisErr(nil); ok || ret != nil {
+		t.Fatal("TestIsRedisErr fail")
+	}
+	if ret, ok := IsRedisErr(errors.New("other")); ok || ret != nil {
+		t.Fatal("TestIsRedisErr fail")
+	}
+	if ret, ok := IsRedisErr(&RedisError{typ: '-'}); !ok || ret.typ != '-' {
+		t.Fatal("TestIsRedisErr fail")
+	}
+}
+
 //gocyclo:ignore
 func TestRedisResult(t *testing.T) {
 	t.Run("ToInt64", func(t *testing.T) {

